zapcore/entry.go

```diff
@@ -163,6 +164,8 @@ const (
 	WriteThenPanic
 	// WriteThenFatal causes a fatal os.Exit after Write.
 	WriteThenFatal
+	// WriteThenGoexit runs runtime.Goexit after Write.
+	WriteThenGoexit
```

Comment:
Comment on CheckWriteAction states:

> Actions are ordered in increasing severity.

So this probably needs to go before panic.